### PR TITLE
[zh-cn] localize blog: 2023-05-16-kmsv2-beta.md

### PR DESCRIPTION
**Pull request by my-git9**
localize blog: 2023-05-16-kmsv2-beta.md

Fix https://github.com/kubernetes/website/issues/41106

### CODE REVIEW

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+that is stored and managed in a remote KMS. With KMS v1, a new DEK is generated for each encryption.
+With KMS v2, a new DEK is only generated on server startup and when the KMS plugin informs the API
+server that a KEK rotation has occurred.
+-->
+## `v2beta1` 有什么新内容？
+
+KMS 加密驱动使用信封加密方式来加密 etcd 中的数据，使用数据加密密钥（DEK）对数据进行加密。
+DEK 使用在远程 KMS 中存储和管理的密钥加密密钥（KEK）进行加密。
+使用 KMS v1，每次加密都会生成一个新的 DEK。
+使用 KMS v2，只有在服务器启动时以及 KMS 插件通知 API 服务器发生 KEK 轮换时才会生成新的 DEK。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:72]:
```suggestion
使用 KMS v2，只有在服务器启动时且 KMS 插件通知 API 服务器发生 KEK 轮换时才会生成新的 DEK。
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+
+{{% alert title="警告" color="warning" %}}
+<!--
+If you are running virtual machine (VM) based nodes
+that leverage VM state store with this feature, you must not use KMS v2.
+
+With KMS v2, the API server uses AES-GCM with a 12 byte nonce (8 byte atomic counter and 4 bytes
+random data) for encryption. The following issues could occur if the VM is saved and restored:
+-->
+如果你运行的是虚拟机（VM）节点，其中具有此特性的节点使用了 VM 的状态存储，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:82]:
```suggestion
如果你运行的是虚拟机（VM）节点，其中启用此特性的节点使用了 VM 的状态存储，
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+则不得使用 KMS v2。
+
+对于 KMS v2，API 服务器使用带有 12 字节随机数（8 字节原子计数器和 4 字节随机数据）的
+AES-GCM 进行加密。如果保存和恢复虚拟机，可能会出现以下问题：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:86]:
```suggestion
AES-GCM 进行加密。在保存和恢复虚拟机时，可能会出现以下问题：
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+
+<!--
+1. The counter value may be lost or corrupted if the VM is saved in an inconsistent state or
+   restored improperly. This can lead to a situation where the same counter value is used twice,
+   resulting in the same nonce being used for two different messages.
+2. If the VM is restored to a previous state, the counter value may be set back to its previous
+   value, resulting in the same nonce being used again.
+-->
+1. 如果 VM 以不一致的状态保存或不正确地恢复，计数器值可能会丢失或损坏。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:95]:
```suggestion
1. 如果 VM 的保存状态不一致或其恢复不正确，计数器值可能会丢失或损坏。
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+   这可能会导致相同的计数器值被使用两次，导致相同的随机数被用于两个不同的消息。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:96]:
```suggestion
   这可能会导致系统再次使用同一计数器值，进而在两个不同的消息中使用相同的随机数。
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+### 性能改进
+
+在 KMS v2 中，我们对 KMS 加密提供程序的性能进行了重大改进。对于 KMS v1，
+每次加密都会生成一个新的 DEK。这意味着对于每个写入请求，API 服务器都会调用
+KMS 插件以使用远程 KEK 加密 DEK。为避免每个读取请求都会调用 KMS 插件，
+API 服务器必须缓存 DEK。当 API 服务器重新启动时，
+它必须根据缓存大小为 etcd 存储中的每个 DEK 调用 KMS 插件来填充缓存。
+这对 API 服务器来说是一个很大的开销。使用 KMS v2，API 服务器在启动时生成一个 DEK 并将其缓存。
+API 服务器还调用 KMS 插件以使用远程 KEK 加密 DEK。这是启动时和 KEK 轮换时的一次性调用。
+API 服务器然后使用缓存的 DEK 来加密资源。这减少了对 KMS 插件的调用次数，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:280]:
```suggestion
在此之后，API 服务器使用缓存的 DEK 来加密资源。这样做减少了对 KMS 插件的调用次数，
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+<!--
+We conducted a test that created 12k secrets and measured the time taken for the API server to
+encrypt the resources. The metric used was
+[`apiserver_storage_transformation_duration_seconds`](https://kubernetes.io/docs/reference/instrumentation/metrics/).
+For KMS v1, the test was run on a managed Kubernetes v1.25 cluster with 2 nodes. There was no
+additional load on the cluster during the test. For KMS v2, the test was run in the Kubernetes CI
+environment with the following [cluster
+configuration](https://github.com/kubernetes/kubernetes/blob/release-1.27/test/e2e/testing-manifests/auth/encrypt/kind.yaml).
+-->
+我们进行了一项创建 12k 大小 Secret 的测试，并检测了 API
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:292]:
```suggestion
我们进行了一项创建 12,000 个 Secret 的测试，并检测了 API
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+For Kubernetes v1.28, we expect the feature to stay in beta. In the coming releases we want to
+investigate:
+- Cryptographic changes to remove the limitation on VM state store.
+- Kubernetes REST API changes to enable a more robust story around key rotation.
+- Handling undecryptable resources. Refer to the
+  [KEP](https://github.com/kubernetes/enhancements/pull/3927) for details.
+-->
+## 下一步计划
+
+对于 Kubernetes v1.28，我们预计该功能仍处于测试阶段。在即将发布的版本中，我们要调研：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:327]:
```suggestion
对于 Kubernetes v1.28，我们预计该功能仍处于测试阶段。在即将发布的版本中，我们将致力于：
```

File: content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+
+- 修改加密程序以消除对 VM 状态存储的限制。
+- Kubernetes REST API 围绕密钥轮换方向修改以实现更强大的特性。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2023-05-16-kmsv2-beta.md:330]:
```suggestion
- 针对密钥轮换，修改 Kubernetes REST API 以实现更强大的特性。
```

**Comment by windsonsea** [comment]
/lgtm

Need to trigger the ci again.

**Pull request merged**